tests/light/functional_tests/destination_drivers/clickhouse_destination/test_clickhouse_connection_options.py

```diff
@@ -0,0 +1,177 @@
+#
+# As an additional exemption you are allowed to compile & link against the
+# OpenSSL libraries as published by the OpenSSL project. See the file
+# COPYING for details.
+#
+#############################################################################
+import uuid
+
+import pytest
+from axosyslog_light.driver_io.clickhouse.clickhouse_io import ClickhouseClient
+
+clickhouse_options = {
+    "database": "default",
+    "table": "test_table",
+    "user": "default",
+    "password": "'password'",
+    "schema": '"message" "String" => "$MSG"',
+}
+
+
+def test_clickhouse_destination_valid_options_db_run(request, config, syslog_ng, clickhouse_server):
+    custom_input_msg = f"test message {str(uuid.uuid4())}"
+    generator_source = config.create_example_msg_generator_source(num=1, template=f'"{custom_input_msg}"')
+    clickhouse_destination = config.create_clickhouse_destination(**clickhouse_options)
+    config.create_logpath(statements=[generator_source, clickhouse_destination])
+
+    clickhouse_server.start()
+    clickhouse_destination.create_table(clickhouse_options["table"], [("message", "String")])
+    request.addfinalizer(lambda: clickhouse_destination.delete_table())
+
+    syslog_ng.start(config)
+
+    assert clickhouse_destination.get_stats() == {'msg_size_avg': 52, 'eps_last_1h': 0, 'processed': 1, 'eps_last_24h': 0, 'batch_size_max': 52, 'memory_usage': 0, 'dropped': 0, 'queued': 0, 'written': 1, 'eps_since_start': 0, 'msg_size_max': 52, 'batch_size_avg': 52}
+    assert clickhouse_destination.read_logs() == [custom_input_msg]
+
+
+def test_clickhouse_destination_valid_options_db_not_run_and_reconnect(request, config, syslog_ng, clickhouse_server):
+    custom_input_msg = f"test message {str(uuid.uuid4())}"
+    generator_source = config.create_example_msg_generator_source(num=1, template=f'"{custom_input_msg}"')
+    clickhouse_options_copy = clickhouse_options.copy()
+    clickhouse_options_copy.update({"time_reopen": 1})
+    clickhouse_destination = config.create_clickhouse_destination(**clickhouse_options_copy)
+    config.create_logpath(statements=[generator_source, clickhouse_destination])
+
+    syslog_ng.start(config)
+
+    assert clickhouse_destination.get_stats() == {'eps_last_24h': 0, 'processed': 1, 'eps_since_start': 0, 'memory_usage': 452, 'dropped': 0, 'queued': 1, 'written': 0, 'eps_last_1h': 0, 'msg_size_max': 52, 'msg_size_avg': 52, 'batch_size_max': 0, 'batch_size_avg': 0}
+
+    assert syslog_ng.wait_for_message_in_console_log("Message added to ClickHouse batch") != []
+    assert syslog_ng.wait_for_message_in_console_log("ClickHouse server responded with a temporary error status code") != []
+
+    clickhouse_server.start()
+    clickhouse_destination.create_table(clickhouse_options["table"], [("message", "String")])
+    request.addfinalizer(lambda: clickhouse_destination.delete_table())
+
+    syslog_ng.reload(config)  # Reload to connect to the ClickHouse server FIXME???
```

Comment:
`time-reopen()` should handle this, if not, this is a bug. Please do not workaround it as this can happen in the wild, too.

Reply:
Sure, this was my intention too. I have removed the reload() part